src/postamble_minimal.js

```diff
@@ -204,6 +204,25 @@ WebAssembly.instantiate(Module['wasm'], imports).then(function(output) {
   /*** ASM_MODULE_EXPORTS ***/
 #endif
   wasmTable = asm['__indirect_function_table'];
+#if ASSERTIONS
+  assert(wasmTable);
+#endif
+
+#if !EXTERNAL_MEMORY
+  wasmMemory = asm['memory'];
```

Comment:
Does this mean that users cannot implement a function named `memory` in their programs? I would recommend renaming this to `__mem`.

Reply:
I do agree in general, and there as been discussion about this in upstream llvm in the past.

The problem is that the name of the memory import/export is determined my wasm-ld.    We have the `--import-memory` and `--export-memory` flags but the name is currently fixed.   The name `__indirect_function_table` was chosen specifically to avoid such collisions.   IIRC we tried to do this in the past but got push back because it would break existing users.

BTW, this doesn't effect all users who have functions called `memory`,  it only effects users who have a function called `memory` who *also* want to include it in `EXPORTED_FUNCTIONS`.  If the former was true I would be far more worried about this concern.

Reply:
That is indeed true.. still feels a bit sloppy I must admit, would be nice to be strict about namespacing :/

Reply:
Also not that with the current situation we already reserve the name `memory` for the export from JS to WebAssembly.. so users today are unable to export JS functions called "memory".   This change reversed that existing restriction and its not clear to make it makes the situation any worse or better.  

We should attempt to re-name `memory` but I think that goal is mostly orthogonal to this change.